skins/laika/src/store/create_data_persister.ts

```diff
@@ -0,0 +1,111 @@
+import { MutationPayload, Store } from 'vuex';
+import { DataPersistenceRepository } from '@/view_models/DataPersistenceRepository';
+import { DataPersistenceItem, DataPersistenceMutationType } from '@/view_models/DataPersistenceItem';
+import { DataEncryptor, SubtleDataEncryptor } from '@/view_models/DataEncryptor';
+
+export class DataPersister {
+	dataEncryptor: DataEncryptor;
+	repository: DataPersistenceRepository;
+	keyNamespace: string;
+	initialValues: { key: string, value: any }[];
+
+	constructor( dataEncryptor: DataEncryptor, repository: DataPersistenceRepository, keyNamespace: string ) {
+		this.dataEncryptor = dataEncryptor;
+		this.repository = repository;
+		this.keyNamespace = keyNamespace;
+		this.initialValues = [];
+	}
+
+	async save( key: string, data: string ) {
+		const encrypted = await this.dataEncryptor.encrypt( data );
+		if ( encrypted !== undefined ) {
+			this.repository.setItem( `${this.keyNamespace}/${key}`, encrypted );
+		}
+	}
+
+	async load( key: string ) {
+		const data = this.repository.getItem( `${this.keyNamespace}/${key}` );
+
+		if ( !data ) {
+			return null;
+		}
+
+		try {
+			return await this.dataEncryptor.decrypt( data );
+		} catch ( e ) {
+			this.repository.removeItem( `${this.keyNamespace}/${key}` );
+			return null;
+		}
+	}
+
+	getPlugin( items: DataPersistenceItem[] ) {
+		return ( store: Store<any> ) => {
+			store.subscribe( ( mutation: MutationPayload, state: any ) => {
+				const item = items.find( item => item.mutation === mutation.type );
+				if ( item ) {
+					switch ( item.mutationType ) {
+						case DataPersistenceMutationType.VALUE:
+							this.save( item.key, JSON.stringify( mutation.payload ) );
+							break;
+						case DataPersistenceMutationType.KEY_VALUE_PAIR:
+							if ( item.fields.includes( mutation.payload.name ) ) {
+								this.save( mutation.payload.name, JSON.stringify( mutation.payload.value ) );
+							}
+							break;
+					}
+				}
+			} );
+		};
+	}
```

Comment:
These 6 levels of nesting are dizzying, but I don't blame you, I know it's because of the sub-optimal way we're storing our data. I wish there was a better way.

Reply:
I'll flip the if statement and reduce one level at least